crates/mb8/tests/test_rand_syscalls.rs

```diff
@@ -0,0 +1,28 @@
+use mb8::vm::VirtualMachine;
+
+#[test]
+fn test_sys_rand_deterministic() {
+    let bin = include_bytes!("../../../kernel/tests/test_sys_rand.bin");
+
+    let mut vm1 = VirtualMachine::default();
+    vm1.devices.rand().seed(234);
+    vm1.load_rom(bin);
+    vm1.run();
+
+    let mut out1 = [0u8; 16];
+    for i in 0..16 {
+        out1[i] = vm1.devices.read(0x0200 + i as u16);
+    }
+
+    let mut vm2 = VirtualMachine::default();
+    vm2.devices.rand().seed(234);
+    vm2.load_rom(bin);
+    vm2.run();
```

Comment:
It looks like out1 and out2 end up identical because both are filled with zeros. The execution never reaches the randomizer syscall.
Would you mind updating the test so that one of the outputs reads from the randomizer and the other reads from memory? That way we can verify that the data written to memory actually comes from the randomizer